src/util/os_utils.js

```diff
@@ -341,15 +340,68 @@ function get_disk_mount_points() {
         });
 }
 
+//https://github.com/adriano-di-giovanni/node-df/blob/master/lib/parse.js
+function _df_parse(stdout) {
+    return stdout
+            .trim()
+            .split(/\r\n|\r|\n/) // split into rows
+            .slice(1) // strip column headers away
+            .map(row => {
+                const columns = row
+                // one or more whitespaces followed by one or more digits
+                // must be interpreted as column delimiter
+                .replace(/\s+(\d+)/g, '\t$1')
+                // one or more whitespaces followed by a slash
+                // must be interpreted as the last column delimiter
+                .replace(/\s+\//g, '\t/')
+                // split into columns
+                .split(/\t/);
+
+        return {
+            filesystem: columns[0],
+            size: parseInt(columns[1], 10),
+            used: parseInt(columns[2], 10),
+            available: parseInt(columns[3], 10),
+            capacity: parseInt(columns[4], 10) * 0.01,
+            mount: columns.slice(5).join(' '),
+        };
+    });
+}
+
+//https://github.com/adriano-di-giovanni/node-df/blob/master/lib/index.js
+async function _df(file_path, flags = '') {
+    const file_name_regex = /[^-_./a-zA-Z0-9]/g;
+    const flags_argument = '-kP' + flags;
+    const args = [flags_argument];
+    if (file_path) {
+        // CVE-2019-15597 add '\' to special charecters to prevent command injection
+        file_path = file_path.replace(file_name_regex, match => '\\' + match);
+        args.push(file_path);
+    }
+    try {
+    const execFile = util.promisify(child_process.execFile);
+    const { stdout, stderr } = await execFile('/bin/df', args);
+    if (stderr) {
+        dbg.error("df command failed:", stderr);
+        throw new Error(stderr);
+    }
+    return _df_parse(stdout);
+    } catch (err) {
+        dbg.error('df command failed:', err);
+        throw err;
+    }
+}
 
 function get_mount_of_path(file_path) {
     console.log('get_mount_of_path');
-    return P.fromCallback(callback => node_df({
-            file: file_path
-        }, callback))
+    try {
+        return _df(file_path)
         .then(function(drives) {
             return drives && drives[0] && drives[0].mount;
         });
+    } catch (err) {
+        dbg.error(err);
+    }
 }
```

Comment:
we can make this function asynced, and also use optional chaining:
```suggestion
async function get_mount_of_path(file_path) {
  console.log('get_mount_of_path');
  try {
    const drives = await _df(file_path);
    return drives?.[0]?.mount;
  } catch (err) {
    dbg.error(err);
  }
}
```